src/ORM/Hierarchy/MarkedSet.php

```diff
@@ -0,0 +1,805 @@
+<?php
+
+namespace SilverStripe\ORM\Hierarchy;
+
+use InvalidArgumentException;
+use LogicException;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBField;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\View\ArrayData;
+
+/**
+ * Contains a set of hierarchal objects generated from a marking compilation run
```

Comment:
Today I learned:
 - Hierarchal - relating to a hierarch.
 - Hierarchical - relating to a hierarchy.